### PR TITLE
Strip port number from host header

### PR DESCRIPTION
**Pull request by tlawrence**
Excon by default adds the default port to the host header for each request
e.g hostname.example.com:443

Some Cloud Providers use filter rules on HAProxy or similar to control API paths to different backends. In our case the ruleset required will increase if we need to filter for port number as well as host headers where this is not specified. 

rfc2616 specifies that the port number should be appended to the host header if it is not the default port. However, 443 IS the default port for https (as specified in rfc2818) Although I accept that this could be seen as ambiguous given that the rfc2818 refers to HTTP over TLS rather than HTTP as defined in rfc2616.

Basically what I'm trying to say (in a long winded rounadbout way..) is should we make ":omit_default_port" (as supported by Excon) the default for vCloud Director  connections? 
If so I think this PR covers it whilst allowing users to override manually if required.

### CODE REVIEW

**Comment by geemus** [comment]
Seems fine. More broadly I'd like to make not including the port the default in excon itself, in time. I'm afraid it might break a lot of stuff though (aws signing for sure).

**Comment by geemus** [comment]
Thanks!

**Comment by pedantic-git** [comment]
This PR has a typo in it (`unless @connection_option[:omit_default_port]` instead of `unless @connection_options[:omit_default_port]`) that causes initialization of `Fog::Compute::VcloudDirector` to fail in the common case. I can patch this downstream but figured it's something that you'd probably like to fix urgently upstream too.

**Comment by geemus** [comment]
Thanks for the heads up, fixed here: https://github.com/fog/fog/commit/c417f0b12d4c6becd0e6b28f0a2414fd614e7c41

**Comment by pedantic-git** [comment]
Thanks!

**Comment by robinbowes** [comment]
Did you fix the tests too to make sure this doesn't regress? </troll>

**Comment by geemus** [comment]
Not yet, figured getting this in before we forgot was a priority. I'd certainly welcome help on getting tests up to snuff also.

**Comment by tlawrence** [comment]
sorry guys.. my bad!

How come the tests didn't fail?

**Comment by pedantic-git** [comment]
That's a bit of a surprise to me too - they [definitely ran](https://travis-ci.org/fog/fog/jobs/103157595) (search for `VcloudDirector`).

**Comment by robinbowes** [comment]
Oooh, it seems github stripped ```</troll>``` from my original comment, ie. I was being not a little facetious

**Comment by geemus** [comment]
It is such a particular thing, probably just not tested appropriately. Also, since we run tests in mock mode, even if that path was exercised it would take some work to catch it. Maybe more trouble than it's worth to worry about (it seems unlikely the typo would be re-introduced). Still, definitely wish we had better coverage in general.
